okdownload/src/main/java/com/liulishuo/okdownload/core/connection/DownloadUrlConnection.java

```diff
@@ -31,16 +34,46 @@
 public class DownloadUrlConnection implements DownloadConnection, DownloadConnection.Connected {
 
     protected URLConnection connection;
+    private Configuration configuration;
+    private URL url;
+    private RedirectHandler redirectHandler;
+
+    private static final String TAG = "DownloadUrlConnection";
 
     DownloadUrlConnection(URLConnection connection) {
+        this(connection, new RedirectHandler());
+    }
+
+    DownloadUrlConnection(URLConnection connection, RedirectHandler redirectHandler) {
         this.connection = connection;
+        this.url = connection.getURL();
+        this.redirectHandler = redirectHandler;
     }
 
     public DownloadUrlConnection(String originUrl, Configuration configuration) throws IOException {
         this(new URL(originUrl), configuration);
     }
 
     public DownloadUrlConnection(URL url, Configuration configuration) throws IOException {
+        this(url, configuration, new RedirectHandler());
+    }
+
+    public DownloadUrlConnection(
+            URL url,
+            Configuration configuration,
+            RedirectHandler redirectHandler) throws IOException {
+        this.configuration = configuration;
+        this.url = url;
+        this.redirectHandler = redirectHandler;
+        configConnection();
+    }
+
+    public DownloadUrlConnection(String originUrl) throws IOException {
+        this(originUrl, null);
+    }
+
+    void configConnection() throws IOException {
```

Comment:
Nope, I mean `configUrlConnection` because all this operation is related to the connection from `url` and once the `url` is changed we have to `config` it. If you name it only `configConnection` it's not easy to relate changing `url` with config connection.

Reply:
updated